src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -85,19 +89,33 @@ default Optional<T> getOptionalSingle(Event e) {
 	 * <p>
 	 * Do not use this in conditions, use {@link #check(Event, Checker, boolean)} instead.
 	 * 
-	 * @param e The event
+	 * @param event The event the expression is being used in.
 	 * @return An array of values of this expression which must neither be null nor contain nulls, and which must not be an internal array.
 	 */
-	public T[] getArray(final Event e);
+	public T[] getArray(Event event);
+	
+	/**
+	 * Get all the values of this expression and then convert them using the provided getter for easy of access.
+	 * 
+	 * @param <G> The returning type of the converter.
+	 * @param event The event the expression is being used in.
+	 * @param converter The converter to do the convertering to the type.
+	 * @return The values completed by the converter.
+	 */
+	@SuppressWarnings("serial")
+	public default <G> G[] get(Event event, Converter<T, G> converter) {
+		assert converter != null;
+		return Converters.convertUnsafe(getArray(event), new TypeToken<G>(converter.getClass()){}.getRawType(), converter);
```

Comment:
I don't like the usage of TypeToken for this. It is annotated with `@Beta`, meaning it's API is subject to breaking change, and we don't shade the dependency directly, instead relying on it through the server jar. This is explained in the description of the `@Beta` annotation:
![image](https://user-images.githubusercontent.com/29547183/184905856-f3e3acef-d0e0-4fd0-b768-6f62a115fda0.png)

Also, did you try using it with TypeToken? Cause how exactly does TypeToken know whether to pick `T` or `G`?

I think it'd be better to add a `Class<G>` parameter

Reply:
Probably for the version Mojang uses of Guava. TypeToken is definitely not being removed, it's the main way Gson operates and uses TypeToken frequently.

Reply:
I'm not saying Guava will ever be removed, but it can be updated to a newer version at some point which may contain breaking changes (as the `@Beta` annotation sugests). Possibly for this reason, Gson doesn't use Guava's TypeToken (ss from `Gson` class source):
![image](https://user-images.githubusercontent.com/29547183/184910530-87fa9f79-b9f2-4ca7-ba48-a2ebf385470f.png)

Reply:
The getRawType method doesn't have the beta annotation in the latest version.

Reply:
Also, did you try using it with TypeToken? Cause how exactly does TypeToken know whether to pick `T` or `G`?

Spigot 1.13 (assuming we don't include this in 2.6) for example, uses Guava 21.0 (ss from Spigot 1.13 jar, `META-INF/maven/org.spigotmc/spigot-api/pom.xml`):
![image](https://user-images.githubusercontent.com/29547183/184943076-22eeb86d-792a-47e7-91b1-514a0d8aaba6.png)

This version for example doesn't have the constructor you're using (`TypeToken#init(Class)`, or not public at least).

Reply:
I've got it working, I even wrote examples in the main description.

Reply:
I am still against usage of TypeToken here (reasons below), but I'll leave that up to other devs to decide.

Not having explicit type specification for arrays may be dangerous, for example:
```java
Number[] nums = stringExpression.getArray(e, t -> {
	if (t.isEmpty())
		return 0;
	else
		return 1;
});
```
One might expect the returned array to be a `Number` array, but it's actually an `Integer` array, which means if it's followed by something like `nums[0] = 1.0;` it'll throw an exception.
If you have an argument to explicitly state the type of the array, this confusion will disappear.

Besides I generally don't like using a 3rd party util like TypeToken for this (especially marked as beta, even though there is a compatible version out that isn't beta anymore), but that could just be me.

Reply:
I don't think something like TypeToken is really necessary here. You should do one of the following maybe:
a. Accept a parameter for Class<G>
b. Instead of taking in Converter take in ConverterInfo
c. Pikachu's comment - is this a necessary addition?

Reply:
TypeToken is widely used as stated, placing a Class parameter is old and outdated when Java is capable of grabbing generics now.

Using a ConverterInfo is a weird API addition, no where in the API does an addon developer need to come into contact with a ConverterInfo. That would be an extra instance the developer would need to make.

Yes this is necessary. When it comes to multiple types, Skript does not have proper API handles for collecting back to an Array.